test/test_markdown_renderer.py

```diff
@@ -127,11 +127,11 @@ def test_numbered_list(self):
         ]
         output = self.roundtrip(input)
         expected = [
-            "22) *emphasized list item*\n",
+            "22)  *emphasized list item*\n",
             "96) \n",
             "128) here begins a nested list.\n",
             "     + apples\n",
-            "     + bananas\n",
+            "     +  bananas\n",
```

Comment:
OK, I will change the `input` (line 122, 123 and 126) as the following
```
-  22)  *emphasized list item*\n
+  22) *emphasized list item*\n

-  96)\n
+  96) \n

-       +  bananas\n
+       + bananas\n
```

I prefer an extra indentation setting like `MarkdownRenderer.max_line_length` to format markdown code.

Reply:
Thank you.

> I prefer an extra indentation setting like MarkdownRenderer.max_line_length to format Markdown code,
and a divided test class like TestMarkdownFormatting to test the formatting function.

This took me a while. 😅 So I expect, because you closed this thread, this is just a note describing the as-is state, i.e. there is this `TestMarkdownFormatting` class that tests the `MarkdownRenderer.max_line_length` setting, right?

Or maybe, like me now, you were also thinking about possibly introducing an optional `boolean` flag passed to the renderer's constructor, so that client code can choose whether to "normalize" the spaces after the leader to 1 space, or not (probably `false` by default)? :)

Reply:
You almost get my point.
Thank for your solution to the thought.
But to clarify, this is a new issue about normalizing and not part of this PR.
Your solution is more workable and simpler than mine to implement it and cares more about performance. 👍

My draft:
I introduce an optional indentation setting passed to the Markdown renderer's constructor.
The setting is an integer and defaults to `None` to not reflow the original lines.
The setting accepts `2` or `4` as input.
For list item, it is used to calculate the spaces after the list marker.
If the setting minus the length of the list marker is greater than 0, 
then it decides the length of the spaces after the list marker.
Otherwise, there is always 1 space after the list marker.

Maybe the Markdown renderer applies the setting to all block tokens which have indentation.
I haven't read CommonMark Spec 0.30 completely yet, 
so I'm not sure about the feasibility of this.

Reply:
OK, I agree that a somewhat more universal solution for indentation and its normalization should go to a separate issue / PR - so don't hesitate to create one. I will be glad if you describe your idea more in depth there.

Note that I'm not expert on the CommonMark Spec either, I rather always just study only what is necessary in order to review another issue or pull request. ;)